src/dbus/mpris.rs

```diff
@@ -145,11 +146,44 @@ impl SpotMprisPlayer {
     }
 
     pub fn seek(&self, Offset: i64) -> Result<()> {
-        Ok(())
+        if !self.can_seek() {
```

Comment:
I'd prefer if calls to can_seek were replaced with `self.state.current_track().is_some()` or a dedicated non-pub fn, since can_seek is the public interface used by dbus clients, and the macros at play here could have unintended side effects!